connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -284,7 +284,8 @@ public ConnectorStateInfo connectorStatus(String connName) {
 
         Collections.sort(taskStates);
 
-        Map<String, String> conf = config(connName);
```

Comment:
we are removing the single use of this method. 

Either we should remove it completely (including its implementations) or we could choose to extend it to return the raw config upon request. Eg. change it to `protected abstract Map<String, String> config(String connName, boolean returnRawConfig);` 

`config(connName, true)` still won't have a use but maybe that's an ok exception for now. Or we can change it to `rawConfig(String connName)`. 

See my comment below about whether we need to create a shallow copy of the config cluster state with the `snapshot` method.